### ISSUE TITLE
[Feature] Support inline comments to mark stable dependencies in `react-hooks/exhaustive-deps`

### ISSUE THREAD

**Issue by luidmidev** [yannickcr/eslint-plugin-react#3935]
## Summary

Currently, ESLint's `react-hooks/exhaustive-deps` rule has hardcoded stable dependencies like `setState` from `useState` and `dispatch` from `useReducer`. However, there's no way for developers to mark additional stable dependencies from third-party hooks or custom hooks that guarantee stable references.

## Problem

Many popular libraries like `react-hook-form`, `@tanstack/react-query`, and others return stable references that don't need to be included in dependency arrays. Currently, developers have to either:

1. Disable the rule entirely with `eslint-disable-next-line`
2. Include stable dependencies unnecessarily 
3. Live with false positive warnings

This creates noise and reduces the effectiveness of the exhaustive-deps rule.

## Proposed Solution

Allow developers to mark specific variables as stable using inline comments during variable assignment:

```javascript
const { 
  handleSubmit, // eslint-stable-dep
  register,     // eslint-stable-dep  
  reset         // eslint-stable-dep
} = useForm();

// Or in destructuring
const { 
  handleSubmit: stableHandleSubmit, // eslint-stable-dep
  register: stableRegister          // eslint-stable-dep
} = useForm();

// Or in regular assignment
const stableCallback = useMyHook(); // eslint-stable-dep
```

### react-hook-form
```javascript
const { 
  handleSubmit, // eslint-stable-dep
  register,     // eslint-stable-dep
  reset         // eslint-stable-dep
} = useForm();

useEffect(() => {
  handleSubmit(onSubmit)();
}, []); // ESLint would not require handleSubmit in deps
```

### Custom hooks with stable returns
```javascript
const stableCallback = useMyCustomHook(); // eslint-stable-dep

useEffect(() => {
  stableCallback();
}, []); // ESLint would not require stableCallback in deps
```

## Alternative Solutions Considered

1. **Configuration-based approach**: Adding rule configuration would require global setup and less granular control
2. **additionalHooks**: This makes ESLint analyze MORE hooks, not treat dependencies as stable
3. **Custom wrappers**: Creates unnecessary code complexity
4. **Selective disable comments**: Reduces rule effectiveness and creates maintenance overhead

## Benefits

- **Granular control**: Mark only specific variables as stable, exactly where needed
- **No configuration needed**: Works immediately without ESLint config changes
- **Self-documenting**: The comment clearly indicates developer intent
- **Maintains rule effectiveness**: Only affects specifically marked dependencies
- **Simple implementation**: Leverages existing ESLint comment parsing infrastructure
- **Backward compatible**: Doesn't affect existing code

Thanks
## Implementation Notes

The feature could work by:
1. Parsing inline comments during variable assignment/destructuring
2. Storing marked variables in a scope-aware registry
3. Excluding marked variables from dependency requirements
4. Supporting various assignment patterns (destructuring, regular assignment, etc.)

This approach is more intuitive than configuration-based solutions and provides the exact granularity developers need.

---

**Is this something the maintainers would be open to? I'd be happy to contribute a PR if this direction looks good.**

**Comment by luidmidev** [comment]
I just realized that the `react-hooks/exhaustive-deps` rule is actually part of the [`eslint-plugin-react-hooks`](https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/src/rules/ExhaustiveDeps.ts) package in the main React repository — not this one (`eslint-plugin-react`).

Would it make more sense to propose this enhancement directly in the React repo instead? I’d be happy to move the discussion or open a new issue there if that’s the appropriate place.

Or is there something I might be missing or overlooking about how this rule is maintained?

Thanks!
